Yosemite/Yosemite/Tools/Stats/LeaderboardStatsConverter.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+import class Aztec.HTMLParser
+import struct Aztec.Element
+
+/// Helper functions to  convert a `Leaderboard` type into a `TopEearnerStat`
+///
+struct LeaderboardStatsConverter {
+
+    /// Returns the ids of the products associated with a top products leaderboar
+    ///
+    static func topProductsIDs(from topProducts: Leaderboard) -> [Int64] {
+        return topProducts.rows.compactMap {
+            Self.infeerProductID(fromHTMLString: $0.subject.display)
+        }
+    }
+
+    /// Indicates what products are missing from a leaderboard in an array of stored products
+    ///
+    static func missingProductsIDs(from topProducts: Leaderboard, in storedProducts: [Product]) -> [Int64] {
+
+        // Get the top products IDs
+        let topProductsIDs = Self.topProductsIDs(from: topProducts)
+
+        // Get the stored products IDs
+        let storedProductsIDs = storedProducts.map { $0.productID }
+
+        // Figure out which top products IDs are missing from store products IDs
+        let missingIDs = Set(topProductsIDs).subtracting(storedProductsIDs)
+        return Array(missingIDs)
+    }
+
+    /// Converts a the`leaderboard rows(top products)` into an array `TopEarnerStatsItem` using an array of stored products to match
+    ///
+    static func topEearnerStatItems(from topProducts: Leaderboard, using storedProducts: [Product]) -> [TopEarnerStatsItem] {
```

Comment:
super nit: maybe "*Stat**s**Items" to match the return type `[TopEarnerStatsItem]`?